saleor/order/migrations/0198_set_draft_base_price_expire_at.py

```diff
@@ -0,0 +1,52 @@
+# Generated by Django 4.2.16 on 2025-03-05 11:21
+from datetime import timedelta
+
+from django.db import migrations, transaction
+from django.db.models import Exists, OuterRef
+from django.utils import timezone
+
+# The batch of size 250 takes ~0.2 second and consumes ~20MB memory at peak
+BATCH_SIZE = 250
+DEFAULT_EXPIRE_TIME = 24
+
+
+def set_base_price_expire_time(apps, _schema_editor):
+    Order = apps.get_model("order", "Order")
+    OrderLine = apps.get_model("order", "OrderLine")
+    orders = Order.objects.filter(status="draft")
+    qs = (
+        OrderLine.objects.filter(Exists(orders.filter(pk=OuterRef("order_id"))))
+        .filter(draft_base_price_expire_at__isnull=True)
+        .order_by("pk")
+    )
+    now = timezone.now()
+    expire_time = now + timedelta(hours=DEFAULT_EXPIRE_TIME)
+    for ids in queryset_in_batches(qs):
+        order_lines = OrderLine.objects.filter(id__in=ids)
+        with transaction.atomic():
+            _order_lines_lock = list(order_lines.select_for_update(of=(["self"])))
```

Comment:
You need to have the same ordering as in the `saleor.order.utils.order_lines_qs_select_for_update`.

Reply:
Added ordering and delegated the data migration to celery task